cppcache/src/util/concurrent/spinlock_mutex.hpp

```diff
@@ -22,21 +22,21 @@
 
 #include <atomic>
 
+#include "apache-geode_export.h"
+
 namespace apache {
 namespace geode {
 namespace util {
 namespace concurrent {
 
-class spinlock_mutex final {
+class APACHE_GEODE_EXPORT spinlock_mutex final {
```

Comment:
Why is this being exported?

Reply:
::sigh:: This is one of my gripes - if you look, you'll find some +30 classes being exported from the cppcache/src directory. Why are any of them exported? It turns out, all our integration tests rely heavily on the private implementation of the native client, either indirectly or directly. That means a fair number of those tests are actually unit tests, and the rest are innocent victims. They were exported because someone wrote a test in the integration-test framework and didn't understand why it wouldn't compile, so they just exported it and the problem magically went away, not understanding why it is this way.

We're going to make a pass and fix all the tests, a combination of reorganizing and breaking those dependencies. For now, this will have to do, and we wouldn't have known this problem existed with this class until I did this.

Reply:
I understand why all the previous classes had to be exported. The old integration tests are a mess. This class however only has a unit test and does not need to be exported.